game-app/game-core/src/main/java/games/strategy/engine/posted/game/pbf/NodeBbForumPoster.java

```diff
@@ -157,12 +168,18 @@ private String uploadSaveGame(
       if (status == HttpURLConnection.HTTP_OK) {
         final String json = EntityUtils.toString(response.getEntity());
         try {
-          final String url = (String) YamlReader.readMap(json).get("url");
-          return "\n[Savegame](" + url + ")";
+          final Map<String, Object> jsonObject = YamlReader.readMap(json);
+          final Map<?, ?> responseObject = (Map<?, ?>) jsonObject.get("response");
+          // This is a temporary hack to handle old versions of nodeBB forum json
+          if (responseObject == null) {
+            return (String) jsonObject.get("url");
+          }
+          final List<?> images = (List<?>) Preconditions.checkNotNull(responseObject.get("images"));
+          final Map<?, ?> imageObject = (Map<?, ?>) images.get(0);
+          return (String) Preconditions.checkNotNull(imageObject.get("url"));
         } catch (final Exception e) {
           // This is a temporary hack to handle old versions of nodeBB forum json
-          final String url = (String) YamlReader.readList(json).get(0).get("url");
-          return "\n[Savegame](" + url + ")";
+          return (String) Preconditions.checkNotNull(YamlReader.readList(json).get(0).get("url"));
```

Comment:
We previously had this being returned `return "\n[Savegame](" + url + ")";` - is the format of the return value different for 'old versions', why is the '[savegame]' string no longer returned?